### DISCUSSION TITLE
Is pockethost meant to be slow?

### DISCUSSION THREAD

**Discussion by snorresovold** [pockethost/pockethost#181] [Q&A]
I am trying pockethost for a school project and it's feeling very slow.
Is there a better way to host this for a faster experience? I don't think I could use this performance in a production app.

**Comment by snorresovold** [comment]
An average request is about 1.5sec btw

**Comment by benallfree** [comment]
@snorresovold That does sound really slow. 

What country are you in?

Can you post the chrome or Firefox network debugging so we can see exactly which requests are taking so long?

**Reply by snorresovold** [reply]
I have my project here: https://sveltebase.vercel.app/
And here is my request
![image](https://user-images.githubusercontent.com/73354016/218071094-95e9cf96-f3bb-46e7-ac17-86c84d6034e7.png)

**Reply by benallfree** [reply]
I see the initial hit taking ~1.5s, but then subsequent requests take ~300ms which is about expected for PocketBase.

<img width="1109" alt="image" src="https://user-images.githubusercontent.com/1068356/218090738-26831afc-0317-4ae3-bb11-d948535cf981.png">

The initial 1.5s happens when your instance is not running and needs to be started. PocketHost does this to conserve resources. When we introduce paid plans, we will be able to leave your instance running all the time. 

That said, I have been seeing a 500ms cold-start penalty, not 1.2s. So that's something I might check into.

**Comment by Mellbin** [comment]
Adding to this, where is PocketHost hosted?
Is there datacenters in US and EU or only US?
Maybe there should be a possibility to select where you want it to be hosted.

**Reply by snorresovold** [reply]
Depending on the cloud provider used this is a possibility.

I think it would be very useful to have this feature on some priced tiers when (if) pocketbase reaches 1.0.

**Comment by bloor** [comment]
I have an instance for a project hosted on a 5$ VPS (cheapest I could get). But hosted in same country as me.
The actual Pocketbase instance is even faster if ran on localhost, the actual queries inside pocketbase instance are < 10ms. The delay is mostly from server location.

![image](https://user-images.githubusercontent.com/1420653/219943886-07429980-3711-4ac4-882c-0453f4efaa57.png)

**Reply by benallfree** [reply]
Yes, this is true as long as both your data and your PocketBase instance reside in the same location.
